### ISSUE TITLE
Upload source to PyPI

### ISSUE THREAD

**Issue by mjwen** [ipcamit/colabfit-descriptor-library#4]
Currently, only Linux wheel is available, no source distribution for wheels for other systems. 

We at least should upload source to PyPI. I guess 
```
python setup.py sdist
wine upload dist/*
```
might be it? 

Alternatively, A CI can also be created to automate it; similar to: https://github.com/openkim/kliff/blob/main/.github/workflows/publish.yml

Longer term, we might want to automate the building of wheels for other systems. This seems promising: https://github.com/pypa/cibuildwheel

**Comment by ipcamit** [comment]
Yes, macOS is bit tricky. Problem is the compiler chain. Libdescriptor uses Enzyme library for differentiation, so for macOS version, I would need to compile it on macOS. I have an old intel based mac, but no access to new ARM silicon. So need to think about the solution. In theory Rosetta stone + macOS should work, but cannot test it yet.

**Comment by mjwen** [comment]
Got it. I guess even if you put a source list on PyPI, a user should not be able to do  `pip install libdescriptor` to compile it because of Enzyme.

**Comment by ipcamit** [comment]
Given the prevalence of the Apple Silicon, I think this will be priority for me. Let me see if I can cross compile or find appropriate services to try.

**Comment by mjwen** [comment]
https://github.com/pypa/cibuildwheel might give it a try. It provides all kinds of environments, including ARM Mac.
